baton/hub.go

```diff
@@ -2,94 +2,86 @@ package baton
 
 import (
 	"encoding/json"
-	"fmt"
 	"log"
-)
 
-type command struct {
-	Module string      `json:"module"`
-	Call   string      `json:"call"`
-	ID     string      `json:"id"`
-	Body   interface{} `json:"body"`
-}
+	"github.com/hackedu/maestro/baton/commands"
+)
 
 type rawMsg struct {
 	conn conn
 	data []byte
 }
 
-type msg struct {
-	conn conn
-	cmd  command
-}
-
 type hub struct {
-	conns      map[conn]struct{}
+	conns      map[conn][]string
+	ids        map[string]conn
```

Comment:
`CommandID` is a good idea.

Looking up connections allows for clean callbacks.
```
case outMsg := <-h.send:
	c, ok := h.ids[outMsg.ID]
	log.Println("Hub: Message for ", outMsg.ID)
	if !ok {
		log.Println("Hub:", outMsg.ID, "is associated with a disconnected client.")
		break
	}
	log.Println(outMsg)
	bytes, err := json.Marshal(outMsg)
	if err != nil {
		log.Println("Hub: Error marshaling commands.Command into JSON")
		log.Println("Hub:", err)
		break
	}
	c.send <- bytes
}
```